File: FSharp/SimpleWorkflowEngine.Core/Step.fs
namespace SimpleWorkflowEngine.Core

type Step = {
    Id:string
    ActivationOperand:LogicOperand
    State:StepState
}
with 
    member me.IsActive with get() = me.State = StepState.Active
    member me.IsCompleted with get() = me.State = StepState.Completed
    member me.SetState(state) = {me with State = state}

File: CSharp/SimpleWorkflowEngine.Core/Step.cs
using System;
using JetBrains.Annotations;
using Volante;

namespace SimpleWorkflowEngine.Core
{
    public class Step : Persistent
    {
        protected internal Step()
        {
        }

        public Step([NotNull] string id, LogicOperand activationOperand, StepState state)
        {
            if (id == null) throw new ArgumentNullException(nameof(id));
            Id = id;
            ActivationOperand = activationOperand;
            State = state;
        }

        public string Id { get; }

        public StepState State { get; }

        public LogicOperand ActivationOperand { get; }

        public bool IsActive() => State == StepState.Active;

        public bool IsCompleted() => State == StepState.Completed;
    }
}